### PR TITLE
ENH: Improve report usability

### PR DESCRIPTION
**Pull request by larsoner**
I'm using `mne.Report` for some data analyses and hit some limitations. This PR proposes a few different things, quoting the changelog entry:
```
- :meth:`mne.Report.add_projs` can now plot with :func:`mne.viz.plot_projs_joint` rather than :func:`mne.viz.plot_projs_topomap`
- :class:`mne.Report` now has attributes ``img_max_width`` and ``img_max_res`` that can be used to control image scaling.
- :class:`mne.Report` now has an attribute ``collapse`` that allows collapsing sections and/or subsections by default.
- :func:`mne.viz.plot_head_positions` now has a ``totals=True`` option to show the total distance and angle of the head.
```
The first is a fairly straightforward improvement to `add_projs` to make use of `plot_projs_joint` instead of plain `plot_projs_topomap` when you want to do so.

Next is adding a few attributes to `mne.Report` to control behavior of 1) image scaling, and 2) section collapsing. I never liked that I couldn't get full-res images using the default functions like `add_raw` or `add_projs`, now you can set `report.img_max_width = None` to stop it from shrinking those plots, e.g. on main (you'll need to open separately to see the downscaling quality issue):

Screenshot from 2024-10-17 16-49-31

On this PR you get full res (up to the browser max-width 100% limitation):

Screenshot from 2024-10-17 16-56-37

And for section collapsing, I added support for collapsing sections and/or subsections by setting `report.collapse = ["section", "subsection"]`. I think this is a good enough start, but I think it's also future compatible with potentially also supporting regex-matching to HTML subsection titles for example.

Finally, it adds an option to `plot_head_positions` to summarize the distance and rotation in terms of the vector magnitude and angle. The distance is nice, but the big gain I think is the angle as it's a lot more human-readable than the quats themselves, e.g.:

image

Draft mode since I need to add tests for all of this stuff and modify some examples but I figured I'd open this to get the ball rolling. I could split these off if it would help review, but happy to keep them together if review will be doable with them all in one place. (Or at least the report stuff together in one PR, and the head pos stuff in another.)

### CODE REVIEW

File: doc/changes/devel/12901.bugfix.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+:class:`mne.Report` HDF5 files are now written in ``mode='a`` (append) to allow users to store other data in the HDF5 files, by `Eric Larson`_.
```
Comment (larsoner) [inline doc/changes/devel/12901.bugfix.rst:1]: Forgot about this change, it's pretty straightforward I think -- use `mode="a"` to make sure you can write companion data to the same HDF5 file without wiping it out. The report is saved in the root-level key `mnepython` so it's easy to use `h5io.write_hdf5(fid, whatever, "myrootkey")` if you've done `fid = h5py.File(..., "a")` already.

File: mne/report/report.py [hunk 2]
```diff
@@ -1660,7 +1768,6 @@ def _add_ica_overlay(self, *, ica, inst, image_format, section, tags, replace):
 
         fig = ica.plot_overlay(inst=inst_, show=False, on_baseline="reapply")
         del inst_
-        _constrain_fig_resolution(fig, max_width=MAX_IMG_WIDTH, max_res=MAX_IMG_RES)
```
Comment (larsoner) [inline mne/report/report.py:1663]: DRYed the code by making `_fig_to_img` always do the constraining rather than calling this all over the place

**Comment by larsoner** [comment]
Okay tests added, this one should be good to go!

File: doc/changes/devel/12901.bugfix.rst (hunk 1 above)
Comment (drammock) [reply doc/changes/devel/12901.bugfix.rst:1]:
```suggestion
:class:`mne.Report` HDF5 files are now written in ``mode='a'`` (append) to allow users to store other data in the HDF5 files, by `Eric Larson`_.
```

File: mne/report/tests/test_report.py [hunk 3]
```diff
@@ -1068,6 +1078,73 @@ def test_manual_report_2d(tmp_path, invisible_fig):
...
+    # Now let's limit it by max resolution (100 dpi)
+    r = Report(image_format="png")
+    r.img_max_width = None
+    with plt.rc_context(rc={"figure.dpi": 200, "figure.figsize": (10, 10)}):
+        r.add_events(**kwargs)
+    r.save(fname, open_browser=False, overwrite=True)
+    imgs = img_re.findall(fname.read_text(encoding="utf-8"))
+    assert len(imgs) == 2
+    img = plt.imread(BytesIO(base64.b64decode(imgs[1].encode("ascii"))))
+    assert img.shape == (1000, 1000, 3)
```
Comment (drammock) [inline mne/report/tests/test_report.py:1134]:
```suggestion
    assert img.shape == (1000, 1000, 3)  # figure.figsize * Report.img_max_res
```

File: mne/report/tests/test_report.py [hunk 4]
```diff
@@ -1068,6 +1078,73 @@ def test_manual_report_2d(tmp_path, invisible_fig):
...
+    # Now let's do unconstrained
+    r = Report(image_format="png")
+    r.img_max_width = r.img_max_res = None
+    with plt.rc_context(rc={"figure.dpi": 200, "figure.figsize": (10, 10)}):
+        r.add_events(**kwargs)
+    r.save(fname, open_browser=False, overwrite=True)
+    imgs = img_re.findall(fname.read_text(encoding="utf-8"))
+    assert len(imgs) == 2
+    img = plt.imread(BytesIO(base64.b64decode(imgs[1].encode("ascii"))))
+    assert img.shape == (2000, 2000, 3)
```
Comment (drammock) [inline mne/report/tests/test_report.py:1145]:
```suggestion
    assert img.shape == (2000, 2000, 3)  # figure.figsize * figure.dpi
```

File: mne/viz/_3d.py [hunk 5]
```diff
@@ -305,10 +323,18 @@ def plot_head_positions(
 
         if destination is not None:
             vals = np.array(
-                [destination[:, 3], rot_to_quat(destination[:, :3])]
+                [1000 * destination[:3, 3], rot_to_quat(destination[:3, :3])]
```
Comment (drammock) [inline mne/viz/_3d.py:326]: huh, was position previously in meters but labeled as millimeters? oops.

**Review by drammock — APPROVED** [review]
looks good!  My suggestions are super minor, feel free to apply them in with `ci skip` and manually merge. Thanks @larsoner!

File: mne/viz/_3d.py (hunk 5 above)
Comment (larsoner) [reply mne/viz/_3d.py:326]: They were in mm before but it was done by modifying `destination` inplace earlier. So no bugfix just a change in where the multiplication happens!

**Pull request merged**